16-google/bgp.ts

```diff
@@ -0,0 +1,47 @@
+import { Stage, Event, metronome, WrappedStage } from "../../src";
+
+export class BGP extends WrappedStage {
+  public percentDropPackets = 0;
+
+  private lastInboundSampled: number = 0;
+  private lastOutboundSampled: number = 0;
+
+  private inboundCounter: number = 0;
+  private outboundCounter: number = 0;
+
+  private interval = 100;
+
+  constructor(protected wrapped: Stage) {
+    super(wrapped);
+    metronome.setInterval(() => this.resetCounts(), this.interval);
+  }
+
+  async workOn(event: Event): Promise<void> {
+    this.inboundCounter++;
+    await this.wrapped.accept(event);
```

Comment:
Should we also drop packets before calling the dependency? Does it matter for our simulation?